src/psyclone/psyir/backend/visitor.py

```diff
@@ -152,7 +152,35 @@ def __call__(self, node):
         :rtype: str
 
         '''
-        return self._visit(node)
+        if not isinstance(node, Node):
```

Comment:
Please update the docstring with a :raises: for the new TypeError. Please could you also extend it to say that a temporary copy of the tree is constructed and then lowered.

Reply:
Updated the docstring